roadmap/implementers-guide/src/glossary.md

```diff
@@ -2,44 +2,71 @@
+- **Relay Parent:** A block in the relay chain, referred to in a context where work is being done in the context of the state at this block.
+- **Router:** The router module is a meta module that consists of three runtime modules responsible for routing messages between paras and the relay chain. The three separate runtime modules are: Dmp, Ump, Hrmp, each responsible for the respective part of message routing.
+- **Runtime:** The relay-chain state machine.
+- **Runtime Module:** See Module.
+- **Runtime API:** A means for the node-side behavior to access structured information based on the state of a fork of the blockchain.
+- **Secondary Checker:** A validator who has been randomly selected to perform secondary approval checks on a parablock which is pending approval.
+- **Subsystem:** A long-running task which is responsible for carrying out a particular category of work.
+- **UMP:** (Upward Message Passing) A vertical message passing mechanism from a parachain to the relay chain.
+- **Validator:** Specially-selected node in the network who is responsible for validating parachain blocks and issuing attestations about their validity.
+- **Validation Function:** A piece of Wasm code that describes the state-transition function of a parachain.
+- **VMP:** (Vertical Message Passing) A family of mechanisms that are responsible for message exchange between the relay chain and parachains.
+- **XCMP:** (Cross-Chain Message Passing) A type of horizontal message passing (i.e. between parachains) that allows secure message passing directly between parachains and has minimal resource requirements from the relay chain, thus highly scalable.
+
+## PVF
+
+The PVF functionality involves several processes which may be potentially
+confusing:
+
+- **Prechecking:** This is the process of initially checking the PVF when it is
+  first added. We attempt *preparation* of the PVF and make sure it succeeds
+  within a given timeout.
+- **Execution:** This actually executes the PVF. The node may not have the
+  artifact from prechecking, in which case this process also includes a
+  *preparation* job. The timeout for preparation here is more lenient than when
+  prechecking.
+- **Preparation:** This is the process of preparing the WASM blob and includes
+  both *prevalidation* and *compilation*. As prevalidation is pretty minimal
+  right now, preparation mostly consists of compilation. Note that *prechecking*
+  just consists of preparation, whereas *execution* will also prepare the PVF if
+  the artifact is not already found.
+- **Prevalidation:** Right now this just tries to deserialize the binary with
+  parity-wasm. It is a part of *preparation*.
+- **Compilation:** This is the process of compiling a PVF from wasm code to
+  a machine code *artifact*. It is a part of *preparation*.
+- **Artifact:** The compiled machine code for the wasm module.
+
+## See Also
```

Comment:
I think you get a bit carried away with your PVF hat on : ) IOW, I don't think this deserves a separate section in the glossary. This is the whole project glossary, and it's not clear if we can continue to introduce sections for each area of interest (and not sure if there can always be a clear divide).

If we do remove the section, then some overly abstract terms will have to go, specifically compilation and execution. But I won't be too sad about that, I don't think they are absolutely necessary. In my subjective opinion, they are quite understandable without explicit spelling out.

They still do care some additional information, but that information should live somewhere else. E.g. overview of a PVF execution mechanism, which does not exist right now, but seems to be useful.

Reply:
Fair enough, I just wanted to be thorough. :) Cleaned up the glossary.

> They still do care some additional information, but that information should live somewhere else. E.g. overview of a PVF execution mechanism, which does not exist right now, but seems to be useful.

I can do that, but can you clarify where that should go? e.g. its own page, or on the PVF Prechecking page? Should it just be about execution? I added a `Summary` section to the PVF Prechecking page, maybe that's enough.